app/services/user.js

```
@@ -0,0 +1,58 @@
+const userRepository = require('../repositories/user');
+const error = require('../services/error');
+
+class UserService {
+  async getUser(id) {
+    const foundUser = await userRepository.getUser(id);
+    if (!foundUser || foundUser.deletedAt) {
+      error.throwNotFoundError("User not found.");
+    }
+    return foundUser;
+  }
+
+  async createUser(user) {
+    let userExists;
+    try {
+      userExists = await userRepository.getUserByEmail(user.email);
+    } catch (err) {
+      error.throwValidationError("Email is required.");
+    }
+    if (!userExists) {
+      try {
+        return await userRepository.createUser(user);
+      } catch (err) {
+        error.throwValidationError("User data invalid.");
+      }
+    } else {
+      error.throwValidationError("Email already used.");
+    }
+  }
+
+  async deleteUser(id) {
+    await this.getUser(id);
```

You should throw an error if there isn't a user with this id.

---

I'm using the getUser service which does that.